functions/Invoke-DbaBalanceDataFiles.ps1

```diff
@@ -84,113 +92,109 @@ function Invoke-DbaBalanceDataFiles {
 		[switch]$Force
 	)
 	
-	begin {
-		Write-Message -Message "Starting balancing out data files" -Level Verbose
+	process {
 		
-		# Try connecting to the instance
-		Write-Message -Message "Attempting to connect to $SqlInstance" -Level Verbose
-		try {
-			$Server = Connect-SqlInstance -SqlInstance $SqlInstance -SqlCredential $SqlCredential
-		}
-		catch {
-			Stop-Function -Message "Failure" -Category ConnectionError -ErrorRecord $_ -Target $SqlInstance -Continue
-		}
+		Write-Message -Message "Starting balancing out data files" -Level Verbose
 		
-		# Check the database parameter
-		if ($Database) {
-			if ($Database -notin $server.Databases.Name) {
-				Stop-Function -Message "One or more databases cannot be found on instance on instance $SqlInstance" -Target $SqlInstance -Continue
+		foreach ($instance in $SqlInstance) {
+			# Try connecting to the instance
+			Write-Message -Message "Attempting to connect to $instance" -Level Verbose
+			try {
+				$Server = Connect-SqlInstance -SqlInstance $instance -SqlCredential $SqlCredential
+			}
+			catch {
+				Stop-Function -Message "Failure" -Category ConnectionError -ErrorRecord $_ -Target $instance -Continue
 			}
 			
-			$DatabaseCollection = $server.Databases | Where-Object { $_.Name -in $Database }
-		}
-		else {
-			Stop-Function -Message "Please supply a database to balance out" -Target $SqlInstance -Continue
-		}
-		
-		# Get the server version
-		$serverVersion = $server.Version.Major
-		
-		# Check edition of the sql instance
-		if ($RebuildOffline) {
-			Write-Message -Message "Continuing with offline rebuild." -Level Verbose
-		}
-		elseif (-not $RebuildOffline -and ($serverVersion -lt 9 -or (([string]$Server.Edition -notlike "Developer*") -and ($Server.Edition -notlike "Enterprise*")))) {
-			# Set up the confirm part
-			$message = "The server does not support online rebuilds of indexes. `nDo you want to rebuild the indexes offline?"
-			$choiceYes = New-Object System.Management.Automation.Host.ChoiceDescription "&Yes", "Answer Yes."
-			$choiceNo = New-Object System.Management.Automation.Host.ChoiceDescription "&No", "Answer No."
-			$options = [System.Management.Automation.Host.ChoiceDescription[]]($choiceYes, $choiceNo)
-			$result = $host.ui.PromptForChoice($title, $message, $options, 0)
-			
-			# Check the result from the confirm
-			switch ($result) {
-				# If yes
-				0 {
-					# Set the option to generate a full backup
-					Write-Message -Message "Continuing with offline rebuild." -Level Verbose
-					
-					[bool]$supportOnlineRebuild = $false
-				}
-				1 {
-					Stop-Function -Message "You chose to not allow offline rebuilds of indexes. Use -RebuildOffline" -Target $SqlInstance
-					return
+			# Check the database parameter
+			if ($Database) {
+				if ($Database -notin $server.Databases.Name) {
+					Stop-Function -Message "One or more databases cannot be found on instance on instance $instance" -Target $instance -Continue
```

Comment:
OH! I like this check...we should add this to the other commands...it will clean up errors of databases not existing.

Need to update the template with this example.